adadamp/_dist.py

```diff
@@ -311,3 +312,46 @@ def _get_gradients(
         grads = [grad_fn(idx=idx) for idx in worker_idxs]
         out = dask.compute(*grads)
         return out
+
+class DaskBaseDamper2(DaskBaseDamper):
+    
+    def fit(self, data, target):
+        start = time.time()
+        super().fit(data, target)
+        
+        stat = {
+            "n_data": self.meta_['n_data'],
+            "n_updates": self.meta_['n_updates'],
+            "epoch_time": time.time() - start,
+            "batch_size": self.batch_size_()
+        }
```

Comment:
You'll want to copy these values. This code says that `stat["n_data"]` is the memory address of `self.meta_["n_data"]`. That means when `self.meta_["n_data"]` changes, `self.stats_[0]["n_data"]` will change too.

To avoid that, use this code:

```python 
from copy import deepcopy

stat = {
    "epoch_time": time.time() - start,
    "batch_size": self.batch_size(),
    **deepcopy(self.meta_),
}
```

Reply:
I might include some more meta information in `stat` too:

* Number of epochs/partial_fit calls
* Number of score calls
* Time for each score call.

I see you have time for each partial_fit call already in `stat`.